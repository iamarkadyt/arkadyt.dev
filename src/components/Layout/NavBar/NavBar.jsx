import React from 'react';
import { NavLink, Link, Route, withRouter } from 'react-router-dom';
import { connect } from 'react-redux';

import * as actionTypes from '../../../store/actions';
import { sentenceToURL } from '../../../utils/utils';
import projects from '../../../projects.content';

import GitHub from 'react-icons/lib/fa/github-alt';
import LinkedIn from 'react-icons/lib/fa/linkedin';
import MailTo from 'react-icons/lib/md/mail';
import ArrowBack from 'react-icons/lib/md/arrow-back';
import IconHome from 'react-icons/lib/go/home';

import classes from './NavBar.module.css';

const MyLink = props => {
    if (props.url === undefined) {
        return <Link to='' className={classes.Link} onClick={props.onClick}>
            {props.children}
        </Link>;
    } else if (props.url.indexOf(':') > -1) {
        return <a href={props.url} className={classes.Link} target="_blank">
            {props.children}
        </a>;
    } else {
        return <NavLink to={props.url} className={classes.Link} activeClassName={classes.active}>
            {props.children}
        </NavLink>
    }
}

class NavBar extends React.Component {
    static get staticRoutes() {
        return [ // URL: UI label
            { url: 'https://github.com/arkadyt', ico: <GitHub /> },
            { url: 'https://linkedin.com/in/arkadyt', ico: <LinkedIn /> },
            { url: 'mailto:arkady.titenko@gmail.com', ico: <MailTo /> },
            { url: 'spacer' },
        ];
    }

    render() {
        return <ul className={classes.list}>
            {this.getItems(NavBar.staticRoutes)}
            <li key={-2}>
                <Route path='/(.+)' render={() => {
                    return <MyLink onClick={() => { this.props.history.push('/'); }}>
                        <span className={classes.content}
                            style={{ animation: `${classes.spin} 1s ease-in-out` }}>
                            <ArrowBack />
                        </span>
                    </MyLink>
                }} />
                <Route path='/' exact render={() => {
                    return <MyLink onClick={() => { this.props.closeAppCover(); }}>
                        <span className={classes.content}>
                            <IconHome />
                        </span>
                    </MyLink>
                }} />
            </li>
            <Route path='/projects' render={() => {
                let routes = [];
                for (let item of projects) {
                    routes.push({ url: '/projects/' + sentenceToURL(item.header), ico: 'â€¢' });
                }
                return this.getItems(routes, true);
            }} />
        </ul>;
    }

    getItems(routes, enableSlideIn = false) {
        return routes.map((item, index) => {
            if (item.url.indexOf('spacer') > -1) {
                return <li key={-1} className={classes.spacer}><hr /></li>
            } else {
                return <li key={`${index}-${item.url}`}>
                    <MyLink url={item.url}>
                        <span style={{ animationDelay: `${index * 250}ms` }}
                            className={[classes.content, enableSlideIn ? classes.slide : ''].join(' ')}>
                            {item.ico}
                        </span>
                    </MyLink>
                </li>
            }
        });
    }
}

const mapDispatchToProps = dispatch => {
    return {
        closeAppCover: () => dispatch({ type: actionTypes.SET_COVER_LIFTED, payload: false })
    }
}

export default withRouter(connect(null, mapDispatchToProps)(NavBar));