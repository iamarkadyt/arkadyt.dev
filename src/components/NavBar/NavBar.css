.NavBar-container {
  position: fixed;
  top: 0;
  left: 0;
  width: 5rem;
  height: 100vh;
  display: flex;
  flex-wrap: wrap;
  background-color: #212121;
  justify-content: center;
  align-content: center; }

.NavBar-container li {
  flex-basis: 5rem;
  height: 5rem;
  display: flex;
  justify-content: center;
  align-items: center;
  list-style: none; }

.NavBar-container li:hover {
  background-color: #313131; }

.NavBar-container li > a {
  color: #eee;
  font-size: 3.2rem;
  width: 100%;
  height: 100%;
  display: flex;
  justify-content: center;
  align-items: center; }
