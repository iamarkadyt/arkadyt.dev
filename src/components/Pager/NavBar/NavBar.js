import React, { Component } from 'react';
import { Link } from 'react-router-dom';

import { CoverCtx } from '../../../hocs/withCoverCtx';

import classes from './NavBar.module.css';

class NavBar extends Component {
    render() {
        return <ul className={classes.list}>
            <CoverCtx.Consumer>
                {context => {
                    return <li>
                        <a href='#' 
                            onClick={() => { context.setCoverLifted(false) }}
                            className={classes.Link}>Home</a></li>
                }}
            </CoverCtx.Consumer>
            {Object.keys(this.props.routes).map(url => (
                <li key={url}>
                    <Link to={url} className={classes.Link}>
                        {this.props.routes[url]}
                    </Link>
                </li>
            ))}
        </ul>;
    };
}

export default NavBar;