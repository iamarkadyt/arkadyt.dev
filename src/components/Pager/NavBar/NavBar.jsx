import React from 'react';
import { NavLink } from 'react-router-dom';

import { CoverCtx } from '../../../hocs/withCoverCtx';

import classes from './NavBar.module.css';

const navBar = props => {
    return <ul className={classes.list}>
        <CoverCtx.Consumer>
            {context => {
                return <li> {/* Home button */}
                    <a href=''
                        onClick={e => { e.preventDefault(); context.setCoverLifted(false); }}
                        className={classes.Link}>Home</a></li>
            }}
        </CoverCtx.Consumer>
        {Object.keys(props.routes).map(url => (
            <li key={url}>
                <NavLink to={url} className={classes.Link} activeClassName={classes.active}>
                    {props.routes[url]}
                </NavLink>
            </li>
        ))}
    </ul>;
}

export default navBar;