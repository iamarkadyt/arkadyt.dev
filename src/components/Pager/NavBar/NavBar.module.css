.list {
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
    align-content: center;

    margin: 0;
    padding: 0;

    height: 100%;
    width: 100%;

    list-style: none;
    background-color: #212121;
}

.list li {
    width: 100%;
    height: 50px;
    margin: 5px 0 0 0;
}

.list li:hover {
    background-color: #313131;
}

.list .spacer {
    height: 10px;
    display: flex;
    justify-content: center;
    align-items: center;
}

.list .spacer:hover {
    background-color: #212121;
}

.list .spacer hr {
    border: 1px solid #414141;
    width: 30px;
}

.list .Link {
    display: block;
    width: 100%;
    height: 100%;

    color: #eee;
    text-decoration: none;
    font-size: 2em;

    border-left: 3px solid transparent;
    border-right: 3px solid transparent;
    box-sizing: border-box;
}

.list .Link.active {
    border-right-color: #eee;
}

.list .Link .content {
    display: flex;
    height: 100%;
    width: 100%;
    justify-content: center;
    align-items: center;
}