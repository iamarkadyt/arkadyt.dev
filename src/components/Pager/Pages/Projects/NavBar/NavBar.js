import React from 'react';

import { Link } from 'react-router-dom';
import { FaCircle } from 'react-icons/lib/fa';

import classes from './NavBar.module.css';

class NavBar extends React.Component {
    render() {
        return <ul className={classes.list}>
            {this.props.urls.map(url => (
                <li key={url}>
                    <Link to={url}>
                        <FaCircle />
                    </Link>
                </li>
            ))}
        </ul>;
    }
};

export default NavBar;