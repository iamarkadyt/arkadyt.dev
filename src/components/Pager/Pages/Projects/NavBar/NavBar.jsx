import React from 'react';

import { NavLink } from 'react-router-dom';
import { FaCircle } from 'react-icons/lib/fa';

import classes from './NavBar.module.css';

class NavBar extends React.Component {
    render() {
        return <ul className={classes.list}>
            {this.props.urls.map(url => (
                <li key={url}>
                    <NavLink to={url} activeClassName={classes.active}>
                        <FaCircle />
                    </NavLink>
                </li>
            ))}
        </ul>;
    }
};

export default NavBar;