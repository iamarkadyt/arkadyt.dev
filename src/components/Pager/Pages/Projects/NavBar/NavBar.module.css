.list {
    margin: 0;
    padding: 0;
    list-style: none;

    display: flex;
    justify-content: center;
    align-items: center;

    width: 100%;
    height: 20%;
}

.list a {
    color: white;
    font-size: 2em;
}

.list li {
    padding: 0 0.5vw;
}

.list .active {
    background: orange;
}