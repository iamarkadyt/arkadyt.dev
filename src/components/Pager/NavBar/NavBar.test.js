import React from 'react';
import Enzyme, { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import { MemoryRouter as Router } from 'react-router-dom';
import NavBar from './NavBar';

configure({ adapter: new Adapter() });

describe('<NavBar/> (at \'/\' route)', () => {
    it('matches snapshot.', () => {
        const routes = { '/projects': 'Projects', '/emailme': 'Email Me' }
    
        const tree = renderer.create(
            <Router>
                <NavBar routes={routes} />
            </Router>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    })
})

describe('Home button', () => {
    it('<a> tag responds on click with right parameters.', () => {
        const routes = { '/projects': 'Projects', '/emailme': 'Email Me' }
    
        const mockFunc = jest.fn();
        const component = shallow(<NavBar routes={routes} />);
        // Bypassing React 16.3 Context element manually. Enzyme's find is yet uncapable of this.
        const homeButtonLi = shallow(component.childAt(0).props().children({ setCoverLifted: mockFunc }));
        homeButtonLi.find('a').simulate('click', { preventDefault: () => {} });
        expect(mockFunc).toHaveBeenCalledWith(false); // coverLifted: false;
    })
})