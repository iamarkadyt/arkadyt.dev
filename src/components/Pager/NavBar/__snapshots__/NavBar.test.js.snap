// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar/> (at '/' route) matches snapshot. 1`] = `
<ul
  className="list"
>
  <li>
     
    <a
      className="Link"
      href=""
      onClick={[Function]}
    >
      Home
    </a>
  </li>
  <li>
    <a
      aria-current={null}
      className="Link"
      href="/projects"
      onClick={[Function]}
      style={undefined}
    >
      Projects
    </a>
  </li>
  <li>
    <a
      aria-current={null}
      className="Link"
      href="/emailme"
      onClick={[Function]}
      style={undefined}
    >
      Email Me
    </a>
  </li>
</ul>
`;
